### ISSUE TITLE
I have run into some problems during the Generalizing- Multi Level. 

### ISSUE THREAD

**Issue by Ashraful-Kalam-Abid** [intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset#3]
![image](https://user-images.githubusercontent.com/68754420/122912078-c5ebca80-d379-11eb-9c98-f8d1d890cfde.png)
I am getting this error when i am trying to run you repo on my google colab. Will you please tell me how to fix this issue?

**Comment by iamollas** [comment]
> ![image](https://user-images.githubusercontent.com/68754420/122912078-c5ebca80-d379-11eb-9c98-f8d1d890cfde.png)
> I am getting this error when i am trying to run you repo on my google colab. Will you please tell me how to fix this issue?

Hello! Can you check with a "!pip freeze" command the version of keras? The version we were working was 2.2.4.

**Comment by Ashraful-Kalam-Abid** [comment]
Do i need to change anything else cause when i installed kerase 2.2.4, it is showing me new error
![image](https://user-images.githubusercontent.com/68754420/123371903-69202800-d5a4-11eb-8916-590c9fc83dba.png)

**Comment by Ashraful-Kalam-Abid** [comment]
![image](https://user-images.githubusercontent.com/68754420/123374005-35470180-d5a8-11eb-8bf8-e2d649b41d6b.png)

**Comment by iamollas** [comment]
Have you cloned to the colab the repo? If yes you need to move the functions to be visible to the directory you are working! These commands should make it work:

!git clone https://github.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset/
!mv '/content/Ethos-Hate-Speech-Dataset/ethos/experiments' '/content/experiments'
!mv '/content/Ethos-Hate-Speech-Dataset/ethos/ethos_data' '/ethos_data'

**Comment by Ashraful-Kalam-Abid** [comment]
@iamollas  yes i have done, as you said.....but now getting this error 
![image](https://user-images.githubusercontent.com/68754420/123388107-43058280-d5ba-11eb-9567-d461afa78703.png)
![image](https://user-images.githubusercontent.com/68754420/123388179-5284cb80-d5ba-11eb-8f9f-f0632bcf0bdc.png)

**Comment by iamollas** [comment]
@Ashraful-Kalam-Abid Okay make a factory reset of the colab and try this:

!git clone https://github.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset/
!mv '/content/Ethos-Hate-Speech-Dataset/ethos/experiments' '/content/experiments'
!mv '/content/Ethos-Hate-Speech-Dataset/ethos/ethos_data' '/ethos_data'
!mv '/content/Ethos-Hate-Speech-Dataset/ethos/hs_data' '/hs_data'
!pip uninstall -y keras-nightly
!pip uninstall -y tensorflow
!pip install -r /content/Ethos-Hate-Speech-Dataset/requirements.txt

If you have problems with the embeddings later , visit the file setC.ipynb which has a cell on how to download, extract and save the embeddings (GloVe).

**Comment by Ashraful-Kalam-Abid** [comment]
Thank you for your reply  @iamollas . 
I will close the issue if it is solved.

**Comment by Ashraful-Kalam-Abid** [comment]
I have not tried for a few days. 
Is this is right direction of keeping embedding file? 


![image](https://user-images.githubusercontent.com/68754420/124391719-c12cfc00-dd13-11eb-94c3-cac82bcff221.png)
![image](https://user-images.githubusercontent.com/68754420/124391771-f8031200-dd13-11eb-843f-8518cd9e9ecf.png)

**Comment by Ashraful-Kalam-Abid** [comment]
@iamollas 
Brother.  I m facing new error.
Can you help?
![image](https://user-images.githubusercontent.com/68754420/124548186-2df8e100-de4f-11eb-8221-76e24b568b73.png)

**Comment by iamollas** [comment]
> @iamollas
> Brother. I m facing new error.
> Can you help?
> ![image](https://user-images.githubusercontent.com/68754420/124548186-2df8e100-de4f-11eb-8221-76e24b568b73.png)

Hello! That's a very common problem. Try downgrading the h5py library to 2.10.0 (pip install 'h5py==2.10.0' --force-reinstall). Then restart the kernel. One thing I noticed is that you are retraining the model (I saw the epochs count/loss/etc). However, in order to replicate the experiment you should skip .fit(params) and just load the weights we are providing. If you train it yourself using fit(params) then you have to load the final.hdf5 model which is saving the best set of weights identified per epoch.

**Comment by Ashraful-Kalam-Abid** [comment]
ok. Thanks
